062-support-tiered-storage.md

```diff
@@ -0,0 +1,169 @@
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for RemoteStorageManager and RemoteLogMetadataManager, users are able to adopt tiered storage feature on Strimzi easily.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+### RemoteStorage
+
+Add a `tieredStorage` config under KafkaClusterSpec, which would support configuration for all tiered storage related configuration.
+
+The CustomTieredStorage class contains configuration for various aspects, including RLM config, RSM specific config, RLMM specific config. 
+
+```
+kafka:
+  tieredStorage:
+      type: custom
+      rlm:
+        # See below detail
+      rsm:
+        # See below detail
+      rlmm:
+        # See below detail
+```
+
+### RLM
+
+The RLM config supports `default` type only.  The config allows users to specify basic configuration for RemoteLogManager in Kafka:
+
+```
+      rlm:
+        threadPoolSize:
+        reader.threads:
+        reader.maxPendingTasks:
+```
+
+### RSM
+
+The RSM config only supports custom type. The config allows users to specify the RSM class information as well as other properties. 
+
+Because the `RemoteStorageManager` dependency is provided by Kafka plugin, the implementation could vary case by case. It’s important to allow user to pass in additional configuration values within the supplied prefix range, thus config field is provided to pass in and generate such properties.
+
+```
+      rsm:
+        type: custom
+        class.name:
+        class.path:  
+        impl.prefix: rsm.config.
+        config:
+          property1:
+          property2:
+          ...
+```
+
+### RLMM
+
+The RLMM config configuration for RemoteLogMetadataManager class. The config allows users to specify RLMM class information as well as other RLMM related propertied. A CustomRLMM class is supported to allow user to supply information on a custom RLMM.
+
+In a custom implementation, Because the dependency is provided by Kafka plugin, the implementation could vary case by case. It’s important to allow user to pass in additional configuration values within the supplied prefix range, thus config field is provided to pass in and generate such properties. Those properties will be concatonated with the prefix value and get applied at Kafka config.
```

Comment:
```suggestion
In a custom implementation, Because the dependency is provided by Kafka plugin, the implementation could vary case by case. It’s important to allow user to pass in additional configuration values within the supplied prefix range, thus config field is provided to pass in and generate such properties. Those properties will be concatenated with the prefix value and get applied at Kafka config.
```